crates/starknet-devnet/tests/test_advancing_time.rs

```diff
@@ -259,18 +257,33 @@ mod advancing_time_tests {
         // wait 1 second
         tokio::time::sleep(time::Duration::from_secs(1)).await;
 
-        // check if after mint timestamp > last block, check if inside buffer limit
-        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
-        let mint_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
-        assert_gt_with_buffer(mint_block.timestamp, empty_block.timestamp);
+        // check if after create block timestamp > last block, check if inside buffer limit
+        devnet.create_block().await.unwrap();
+        let latest_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+        assert_gt_with_buffer(latest_block.timestamp, empty_block.timestamp);
     }
 
     #[tokio::test]
-    async fn set_time_in_future() {
+    async fn set_time_in_past_normal_mode() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        set_time_in_past(&devnet).await;
+    }
+
+    #[tokio::test]
+    async fn set_time_in_past_block_on_demand_mode() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--blocks-on-demand"])
+                .await
+                .expect("Could not start Devnet");
```

Comment:
Here and in other places, the line can be shorter if you just do:
```rust
let devnet = BackgroundDevnet::spawn_with_additional_args(&["--blocks-on-demand"]).await.unwrap();
```

Reply:
remove of redundant : BackgroundDevnet in whole project is here 5916ecc01a0cf4ef00799b29814e9daa662bf93a

Reply:
> "Could not start Devnet" appears in my terminal so I know that I need to build a release

If `.expect(...)` is used, you get:
```
Could not start devnet: DevnetNotStartable
```

If `.unwrap()` is used, you get:
```
called `Result::unwrap()` on an `Err` value: DevnetNotStartable
```

Just for future spawnings of BackgroundDevnet. I'll resolve this conversation.